File: packages/runtime/neft-runtime-ios/Neft/NativeBinding.swift
import Foundation

class NativeBinding {
    var module: String

    init(_ module: String) {
        self.module = module
    }

    private func getFullName(_ name: String) -> String {
        return "\(module)/\(name)"
    }

    func onCall(_ name: String, _ handler: @escaping ([Any?]) -> Void) -> NativeBinding {
        App.getApp().client.addCustomFunction(getFullName(name), function: handler)
        return self
    }

    func pushEvent(_ name: String, args: [Any?]?) {
        App.getApp().client?.pushEvent(getFullName(name), args: args)
    }

    func finalize() {}
}

File: packages/runtime/neft-runtime-android/app/src/main/java/io/neft/client/NativeBinding.java
package io.neft.client;

import android.support.annotation.NonNull;

import io.neft.App;
import io.neft.utils.Consumer;

public class NativeBinding {
    private final static App APP = App.getInstance();
    private final String module;

    public NativeBinding(@NonNull String module) {
        this.module = module;
    }

    private String getFullName(String name) {
        return module + "/" + name;
    }

    public NativeBinding onCall(@NonNull String name, @NonNull Consumer<Object[]> handler) {
        APP.getClient().addCustomFunction(getFullName(name), handler);
        return this;
    }

    public void pushEvent(@NonNull final String name, final Object... args) {
        APP.getActivity().runOnUiThread(new Runnable() {
            @Override
            public void run() {
                APP.getClient().pushEvent(name, args);
            }
        });
    }
}
